doc/modules/grid_search.rst

```diff
@@ -167,6 +171,342 @@ variable that is log-uniformly distributed between ``1e0`` and ``1e3``::
+parameter of :class:`HalvingGridSearchCV`.
+
+Consider a case where the resource is the number of samples, and where we
+have 1000 samples. With ``min_resources=10`` and ``ratio=2`` we are able to
+run 7 iterations, with the following number of samples: ``[10, 20, 40, 80,
+160, 320, 640]``.
+
+If we start with a high number of candidates, we might end up with a lot of
+candidates at the last iteration. On the other hand if we start with a small
+number of candidates, the last iteration might use less than 640 samples
+which is a waste of resources.
+
+In the case of :class:`HalvingRandomSearchCV`, the number of candidates is set
+by default such that the maximum amount of resources is used at the last
+iteration.
+
+Changing the value of ``min_resources`` will impact the number of possible
+iterations, and as a result will also have an effect on the ideal number of
+candidates.
+
+Another consideration when choosing ``min_resources`` is whether or not it
+is easy to discriminate between good and bad candidates with a small amount
+of resources. For example, if you need a lot of samples to distinguish
+between good and bad parameters, a high ``min_resources`` is recommended. On
+the other hand if the distinction is clear even with a small amount of
+samples, then a small ``min_resources`` may be preferable since it would
+speed up the computation.
+
+Notice in the example above that the last iteration does not use the maximum
+amount of resources available: 1000 samples are available, yet only 640 are
+used, at most. By default, ``min_resources`` is set to a specific value such
+that the last iteration uses as many samples as possible. Please see
+:ref:`exhausting_the_resources` for details.
+
+.. _amount_of_resource_and_number_of_candidates:
+
+Amount of resource and number of candidates at each iteration
+-------------------------------------------------------------
+
+The amount of resources ``resource_iter`` (e.g. the number of samples)
+allocated for each candidate at iteration ``i`` is controlled by the
+parameters ``ratio`` and ``min_resources`` as follows::
+
+    resource_iter = ratio**i * min_resources
+
+``min_resources`` is the amount of resources used at the first iteration and
+``ratio`` defines the proportions of candidates that will be selected for
+the next iteration (``ratio`` must be greater than 1)::
+
+    n_candidates_to_keep = n_candidates_at_i // ratio
+
+So in the first iteration, we use ``min_resources`` resources
+``n_candidates`` times. In the second iteration, we use ``min_resources *
+ratio`` resources ``n_candidates // ratio`` times. The third again
+multiplies the resources per candidate and divides the number of candidates.
+This process stops when the maximum amount of resource per candidate is
+reached, or when we have identified the best candidate. The best candidate
+is identified at the iteration that is evaluating `ratio` or less candidates
```

Comment:
I know how it works, but in my mind those two are two different parameters, don't you think?

Reply:
I'm not sure what you're saying.

There's only one parameter here, and it's used to divide the number of candidates for the next iteration. We called it `ratio`, but I'm def open to other names. By definition, it's also used to multiply the number of resources: in theory that could be another value, but that's not SH anymore then.


Note that we don't use `ratio` as a stopping criteria because we *want* to, but because we **have** to:  there is no point for the SH process to continue once we have less than `ratio` candidates. What would we do at the next iteration anyway? we would only be left with 1 candidate, so evaluating this one alone is useless, right?

Reply:
I think what @adrinjalali meant was that ``ratio`` has values >1, while the name suggests they are fractions, right? Maybe ``factor`` would be better? though not sure that's worth the large replace?

Reply:
`ratio` seems to bother a lot of people. Joel suggested `rate`, which I think is good. I also like `factor`. Happy to change for whatever you might prefer?